index.rst

```diff
@@ -42,20 +42,151 @@
+Additionally it may be necessary to provide an explicit config file by name.
+Using standard change-controlled pipeline configurations is preferable to scripts trying to send override configuration on-the-fly since it provides a self-contained way to reproduce the processing.
+
+For a Pipeline to be executed the data ID needs to be converted to a quantum graph suitable for passing to the activator.
+The data ID would usually be the obsid for the data just acquired.
+Furthermore the group ID could be used to obtain all data that have previously been taken related to this observation.
 
 .. note::
 
-   **This technote is not yet published.**
+  No LSST rapid analysis pipelines are required to co-add data incrementally.
+  Pipelines should not co-add data as it arrives and should always defer co-addition until the final observation in the group has been taken.
+
+Single frame processing is the simplest mode whereby the products from the processing are not to be combined with subsequent observations.
+The more complex scenario relates to processing that requires co-addition where the co-addition can only occur at the end of data acquisition.
+
+One proposal for this is to have a chained pipeline approach.
+As data arrives they are processed with the nominated single frame processing variant.
+On completion they register themselves as inputs to a combination pipeline.
+This pipeline does not run automatically but waits to be told that all data have arrived.
+If an observation fails to reduce properly, or fails to pass some metrics threshold it will not register with the combination pipeline.
+
+When all expected data have arrived and been processed on their own, the final combination pipeline will be unblocked and will run to completion.
+Events will be issued on the completion of a single frame processing and on the completion of the combination pipeline.
+Metrics will also be published where appropriate during processing although the mechanism for this has not been defined.
+
+The issues that have to be solved are:
+
+1. How does the pipeline know to start processing a data ID?
+2. Which pipeline should be run for the given data ID?
+3. If the pipeline supports chaining, is that known to the system passing in the data?
+4. Who decides which observations should be combined and how?
+5. How does the system know the final observation arrived such that the combination pipeline can run?
+
+This document will describe two alternate scenarios for answering the above questions.
+The first describes a system entirely controlled by the observing script that is running the observation.
+The second describes a system that is decoupled from the observing system and is entirely driven by data arriving in the system.
+
+To simplify the remainder of this document the name given to the system that forwards processing requests to pipelines will be the "OCS Controlled Pipeline System (OCPS)."
```

Comment:
Should note that this replaces the OCS-Controlled Batch service in LDM-148.

Reply:
We could still call it that although I'm not sure "batch" gives the right connotation in what we are saying below. I wanted something short in the later sections but OCBS could take that role.  That reminds me I should mention what happens if we start backing up and not processing data fast enough.